cmd/minikube/cmd/start.go

```diff
@@ -1419,6 +1425,23 @@ func validateKubernetesVersion(old *config.ClusterConfig) {
 		exitIfNotForced(reason.KubernetesTooOld, "Kubernetes {{.version}} is not supported by this release of minikube", out.V{"version": nvs})
 	}
 
+	if nvs.GT(newestVersion) {
+		k8sReleases, err := github.RecentK8sVersions()
+		if err != nil {
+			exit.Message(reason.KubernetesReleaseFetchFailed, "Failed fetching available Kubernetes releases from GitHub repository. Check your internet connection")
```

Comment:
This needs to have some kind of offline override, when running in an air-gapped installation.

Reply:
How would you design this? We're not able to get the versions offline without hardcoding them in the code. But we can make this feature dependent on a configuration option (basically make a feature flag). The question is, is there already a config flag that specifies that minikube is running on an air-gapped environment? If yes, we'd just have to implement feature flag here. If not, we should probably add a config option? What do you reckon?

Reply:
Possibly re-use the force flag, which is the generic bail-out.

Reply:
`❗  minikube skips various validations when --force is supplied; this may lead to unexpected behavior`

Reply:
So if force flag is enabled, we skip just fetching releases, not the whole validation, right?

Reply:
There needs to be a way to skip all validation, in case GitHub is down or if you want to run a custom kubernetes

Reply:
So essentially you want to wrap this line: https://github.com/kubernetes/minikube/blob/d76259e86f4845bd01fe3e81e65214b9e231ecdb/cmd/minikube/cmd/start.go#L189
with a feature flag based on `--force` ?

Why have somebody not done that yet? 
I don't know if we should go that far in this PR. Validation is good even if we specify the `force` flag. Error will happen anyway but in the other place and with an obscure message if we turn off the validation.

I'd personally wrap just the release fetching inside the validation, because turning off that one particular validation really does matter in particular setting - while we have no internet connection.

Reply:
Also, tagging @medyagh, I think we might want to hear your opinion on that

Reply:
I agree with @afbjorklund we better make it work that if user already downloaded minikube and k8s preload there shouldnt be any need to make outgoing calls
so I suggest doing this
if Major Version is not "1"
then we would do a Github Release check, 
and if that failed we shoudl add a out.BoxedWarnning that the kuberentes verison u asked for doesnt seem to be right

but do not exit and let it continue

and also make sure this check does NOT happen for versions before our current NewestKubernetesVersion